jukebox_radio/streams/models/marker_model.py

```diff
@@ -19,19 +19,29 @@ def serialize(self, marker):
         }
 
 
+class MarkerQuerySet(models.QuerySet):
+    def filter_by_track_and_user(self, track_uuid, user):
+        return (
+            self
+            .filter(
+                user=user,
+                track_id=track_uuid,
+                deleted_at__isnull=True,
+            )
+            .order_by("timestamp_ms")
+        )
+
+
 @pgtrigger.register(
-    pgtrigger.Protect(
-        name="append_only",
-        operation=(pgtrigger.Update | pgtrigger.Delete),
-    )
+    pgtrigger.Protect(name="protect_deletes", operation=pgtrigger.Delete)
```

Comment:
we need update so we can soft delete/ archive.

Reply:
in general, triggers and models are being implemented for minimum needed behavior. later, more defenses could be added.